### PR TITLE
add java sol to cses array description

### PR DESCRIPTION
**Pull request by TruaShamu**

### CODE REVIEW

File: solutions/CSES Array Description.mdx [hunk 1]
```diff
@@ -62,5 +65,77 @@ int main() {
...
+</CPPSection>
+
+<JavaSection>
+
+```java
+
+import java.io.*;
+import java.util.*;
+
+public class cses1746 {
```
Comment (SansPapyrus683) [inline solutions/CSES Array Description.mdx:77]: Rename this to `ArrayDesc` or something else.

File: solutions/CSES Array Description.mdx [hunk 2]
```diff
@@ -62,5 +65,77 @@ int main() {
...
+	public static int MOD = (int) 1e9 + 7;
```
Comment (SansPapyrus683) [inline solutions/CSES Array Description.mdx:78]: `public static final int` instead?

File: solutions/CSES Array Description.mdx [hunk 3]
```diff
@@ -62,5 +65,77 @@ int main() {
...
+	public static void main(String args[]) throws IOException {
+		BufferedReader r;
+		r = new BufferedReader(new InputStreamReader(System.in));
```
Comment (SansPapyrus683) [inline solutions/CSES Array Description.mdx:81]: You can probably condense these two statements into a single line.

File: solutions/CSES Array Description.mdx [hunk 4]
```diff
@@ -62,5 +65,77 @@ int main() {
...
+		StringTokenizer st = new StringTokenizer(r.readLine());
+		int N = Integer.parseInt(st.nextToken());
+		int M = Integer.parseInt(st.nextToken());
+		int[][] dp = new int[N][M + 1];
+
+		/* Process the first element in the array. */
```
Comment (SansPapyrus683) [inline solutions/CSES Array Description.mdx:87]: use `//` instead of that comment format

File: solutions/CSES Array Description.mdx [hunk 5]
```diff
@@ -62,5 +65,77 @@ int main() {
...
+							dp[i][j] += dp[i - 1][k];
+							dp[i][j] %= MOD;
+						}
+					}
+				}
+			} else {
+				/* Find number of ways that the current value is valid. */
+				int[] moves = new int[] {A - 1, A, A + 1};
+				for (int k : moves) {
+					if (1<=k && k <= M) {
```
Comment (SansPapyrus683) [inline solutions/CSES Array Description.mdx:119]: add whitespace between operators plz

File: solutions/CSES Array Description.mdx [hunk 6]
```diff
@@ -62,5 +65,77 @@ int main() {
...
+
+public class cses1746 {
+	public static int MOD = (int) 1e9 + 7;
+	public static void main(String args[]) throws IOException {
+		BufferedReader r;
+		r = new BufferedReader(new InputStreamReader(System.in));
+		StringTokenizer st = new StringTokenizer(r.readLine());
+		int N = Integer.parseInt(st.nextToken());
+		int M = Integer.parseInt(st.nextToken());
+		int[][] dp = new int[N][M + 1];
```
Comment (SansPapyrus683) [inline solutions/CSES Array Description.mdx:85]: perhaps add a short explanation of what this dp array contains
like `dp[i][j] contains (insert whatever here)`

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]
just some formatting issues
also, i'd prefer if you used more intuitive variable names `arr` and `maxVal`, this would make it easier to read

File: solutions/CSES Array Description.mdx [hunk 7]
```diff
@@ -62,5 +65,79 @@ int main() {
...
+		}
+
+		for (int i = 1; i < N; i++) {
+			int A = Integer.parseInt(st.nextToken());
+
+			// A free element.
+			if (A == 0) {
+
+				for (int j = 1; j <= M; j++) {
+					int[] arr = new int[] {j - 1, j, j + 1};
```
Comment (SansPapyrus683) [inline solutions/CSES Array Description.mdx:107]: rename this to something like `neighborVals` or something

File: solutions/CSES Array Description.mdx [hunk 8]
```diff
@@ -62,5 +65,79 @@ int main() {
...
+		StringTokenizer st = new StringTokenizer(r.readLine());
+		int N = Integer.parseInt(st.nextToken());
+		int M = Integer.parseInt(st.nextToken());
+
+		// DP[i][j] is the number of ways
+		// to make a valid array from 0 to i, if the ith value is j.
+		int[][] dp = new int[N][M + 1];
+
+		// Process the first element in the array.
+		st = new StringTokenizer(r.readLine());
```
Comment (SansPapyrus683) [inline solutions/CSES Array Description.mdx:90]: maybe rename this to `arr` so it's obvious that it contains the elements?

File: solutions/CSES Array Description.mdx [hunk 9]
```diff
@@ -62,5 +65,79 @@ int main() {
...
+		//Aggregate the dp values.
+		int ans = 0;
+		for (int j = 1; j <= M; j++) {
+			ans += dp[N - 1][j];
+			ans %= MOD;
+		}
+
+		System.out.println(ans);
+	}
+}
```
Comment (SansPapyrus683) [inline solutions/CSES Array Description.mdx:138]: lines 102, 105, and 135 seem to be unnecessary whitespace

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Review by jessechoe10 — APPROVED** [review]

**Pull request merged**